timelock-impl/src/test/java/com/palantir/atlasdb/timelock/adjudicate/LeaderElectionDurationAccumulatorTest.java

```diff
@@ -0,0 +1,152 @@
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import com.google.common.util.concurrent.Uninterruptibles;
+import com.palantir.common.concurrent.PTExecutors;
+import com.palantir.conjure.java.lib.SafeLong;
+import com.palantir.timelock.feedback.LeaderElectionDuration;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.LongConsumer;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import org.junit.Test;
+
+public class LeaderElectionDurationAccumulatorTest {
+    private static final UUID LEADER_1 = UUID.randomUUID();
+    private static final UUID LEADER_2 = UUID.randomUUID();
+    private static final UUID LEADER_3 = UUID.randomUUID();
+
+    private LongConsumer mockConsumer = mock(LongConsumer.class);
+    private LeaderElectionDurationAccumulator accumulator = new LeaderElectionDurationAccumulator(mockConsumer, 5);
+
+    @Test
+    public void nothingConsumedWithFewUpdates() {
+        leaderElectionResultsWithDurationInRandomOrder(LEADER_1, LEADER_2, 4, 1);
+        leaderElectionResultsWithDurationInRandomOrder(LEADER_2, LEADER_3, 4, 5);
+
+        verifyNoInteractions(mockConsumer);
+    }
+
+    @Test
+    public void consumeAfterFiveUpdatesWithMinDuration() {
+        leaderElectionResultsWithDurationInRandomOrder(LEADER_1, LEADER_2, 5, 1);
+
+        verify(mockConsumer).accept(1L);
+        verifyNoMoreInteractions(mockConsumer);
+    }
+
+    @Test
+    public void consumeOnlyOnceUsingMinFromFirstFiveUpdates() {
+        leaderElectionResultsWithDurationInRandomOrder(LEADER_1, LEADER_2, 5, 15);
+        leaderElectionResultsWithDurationInRandomOrder(LEADER_1, LEADER_2, 10, 1);
+
+        verify(mockConsumer).accept(15L);
+        verifyNoMoreInteractions(mockConsumer);
+    }
+
+    @Test
+    public void consumeOnceForEachLeaderPair() {
+        leaderElectionResultsWithDurationInRandomOrder(LEADER_1, LEADER_2, 5, 10);
+        leaderElectionResultsWithDurationInRandomOrder(LEADER_2, LEADER_3, 5, 15);
+        leaderElectionResultsWithDurationInRandomOrder(LEADER_1, LEADER_2, 5, 1);
+        leaderElectionResultsWithDurationInRandomOrder(LEADER_1, LEADER_3, 5, 7);
+
+        verify(mockConsumer).accept(10L);
+        verify(mockConsumer).accept(15L);
+        verify(mockConsumer).accept(7L);
+        verifyNoMoreInteractions(mockConsumer);
+    }
+
+    @Test
+    @SuppressWarnings("ExecutorSubmitRunnableFutureIgnored")
+    public void testInterleavingNoPause() throws InterruptedException {
+        ExecutorService executorService = PTExecutors.newFixedThreadPool(3);
+        accumulator = new LeaderElectionDurationAccumulator(mockConsumer, 300);
+        executorService.submit(() -> leaderElectionResultsWithDurationInRandomOrder(LEADER_1, LEADER_2, 300, 1));
```

Comment:
The comment above is especially important here, IMO.

Reply:
I think it's still technically okay (assuming each election generates one thing) because the 1L could only have come from the first call and so on, though it is probably clearer in terms of readability to have non-overlapping ranges.